### PR TITLE
tests: add remain ha cases with P2

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
Many cases have been covered before.  This pr add the remaining ha cases for P2.

### What is changed and how it works?
- comment for old cases
- add new cases TICASE-936,946,947,957,970,971,976,977,978,980,983,984,987,988,992,998

### Something should be notice
- Network delay cases TICASE-937,938,939 are not added
- offline-master cases TICASE-943,935 are not added
- upgrade dm cases TICASE-1004,1005,1006 are not added
- TICASE-948,949,950,965 are hard to add in CI
- TICASE-936, DM currently not support join during network partition. See https://github.com/etcd-io/etcd/issues/10738
- many cases were added base on previous function for speed, no logic and elegance

### Tests 
 - Unit test
 - Integration test

### CODE REVIEW

File: tests/all_mode/run.sh [hunk 1]
```diff
@@ -36,21 +36,13 @@ function test_session_config(){
...
-    
-    run_sql_file $cur/data/db1.increment.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1
-    run_sql_file $cur/data/db2.increment.sql $MYSQL_HOST2 $MYSQL_PORT2 $MYSQL_PASSWORD2
-
-    # but syncer should fail
-    echo "check diff should fail"
-    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml 10 "fail"
-
+    # fail because insert 0 will auto generates the next serial number
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml 10 'fail'
```
Comment (csuzhangxc) [inline tests/all_mode/run.sh:45]: which stage it will fail, _load_ or _sync_?

Comment (GMHDBJD) [reply tests/all_mode/run.sh:45]: load fail because we insert value (0, 'lalala') in db.prepare

File: tests/ha_cases/run.sh [hunk 2]
```diff
@@ -470,60 +560,101 @@ function test_multi_task_reduce_and_restart_worker() {
...
+    echo "[$(date)] <<<<<< finish test_isolate_master_and_worker >>>>>>"
 }
 
 
 function run() {
-    test_join_masters_and_worker               # TICASE-928, 930, 931, 961
-    test_kill_master                           # TICASE-996
-    test_kill_worker                           # TICASE-968, 973, 1002
+    test_join_masters_and_worker               # TICASE-928, 930, 931, 961, 932, 957
+    test_kill_master                           # TICASE-996, 944, 958
```
Comment (csuzhangxc) [inline tests/ha_cases/run.sh:649]: `test_kill_master` seems not _restart_ as `TICASE-944` expecteed.

Comment (csuzhangxc) [reply tests/ha_cases/run.sh:649]: Is there any comments about `TICASE-945`?

File: tests/ha_cases/run.sh [hunk 3]
```diff
@@ -470,60 +560,101 @@ function test_multi_task_reduce_and_restart_worker() {
...
+    test_kill_worker                           # TICASE-968, 973, 1002, 975, 969, 972, 974, 970, 971, 976, 978, 988
```
Comment (csuzhangxc) [inline tests/ha_cases/run.sh:650]: where do we _pause_ and _resume_ task for `TICASE-979` in `test_kill_worker`?

File: tests/ha_cases/run.sh [hunk 4]
```diff
@@ -470,60 +560,101 @@ function test_multi_task_reduce_and_restart_worker() {
...
     test_kill_master_in_sync
     test_kill_worker_in_sync
-    test_standalone_running                    # TICASE-929, 959, 960, 967
+    test_standalone_running                    # TICASE-929, 959, 960, 967, 977, 980
     test_pause_task                            # TICASE-990
-    test_multi_task_reduce_and_restart_worker  # TICASE-968, 994, 995
-    test_isolate_master                        # TICASE-934, 935
-    test_stop_task                             # TICASE-991
+    test_multi_task_reduce_and_restart_worker  # TICASE-968, 994, 995, 964, 966, 979, 981, 982, 985, 986, 989, 993
```
Comment (csuzhangxc) [inline tests/ha_cases/run.sh:655]: Do we have added `TICASE-983`?

File: tests/ha_cases/run.sh [hunk 5]
```diff
@@ -470,60 +560,101 @@ function test_multi_task_reduce_and_restart_worker() {
...
+    test_isolate_master_and_worker             # TICASE-934, 935, 936, 987, 992, 998
```
Comment (csuzhangxc) [inline tests/ha_cases/run.sh:656]: Is there any comments about `TICASE-999`?

File: tests/ha_cases/run.sh [hunk 6]
```diff
@@ -299,6 +345,48 @@ function test_standalone_running() {
...
+    
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "stop-task test2"\
+        "\"result\": true" 1
+    
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "start-task $cur/conf/standalone-task2.yaml"\
+        "\"result\": false" 1
+
+    # test running, test2 fail
```
[the remark this answers is not in the document]
Comment (GMHDBJD) [reply tests/ha_cases/run.sh]: address in 75c8977

File: tests/ha_master/run.sh [hunk 7]
```diff
@@ -159,6 +159,26 @@ function test_list_member() {
         "list-member --master" \
         "\"alive\": true" 5
     
+    
+    # restart follower
```
[the remark this answers is not in the document]
Comment (GMHDBJD) [reply tests/ha_master/run.sh:163]: both 933 and 943 are cover in 75c8977

File: tests/ha_cases/run.sh (hunk 2 above)
Comment (GMHDBJD) [reply tests/ha_cases/run.sh]: both 944 and 945 have been covered in `test_list_member`, 75c8977 add comments for them.

File: tests/ha_cases/run.sh (hunk 3 above)
Comment (GMHDBJD) [reply tests/ha_cases/run.sh:650]: https://github.com/pingcap/dm/blob/75c897777ab902db6d9552ab8c001dcdfd0a8133/tests/ha_cases/run.sh#L625 here

File: tests/ha_cases/run.sh (hunk 4 above)
Comment (GMHDBJD) [reply tests/ha_cases/run.sh:655]: comment in 75c8977

File: tests/ha_cases/run.sh (hunk 5 above)
Comment (GMHDBJD) [reply tests/ha_cases/run.sh]: I think that test_isolate_master_and_worker already cover it and add comment in 75c8977

File: tests/ha_cases/run.sh (hunk 3 above)
Comment (csuzhangxc) [reply tests/ha_cases/run.sh:650]: If so, should we need to put `977` to L656?

File: tests/ha_cases/run.sh [hunk 8]
```diff
@@ -175,14 +184,36 @@ function test_kill_worker() {
...
     check_port_offline $WORKER3_PORT 20
     rm -rf $WORK_DIR/worker3/relay_log
 
     echo "wait and check task running"
     check_http_alive 127.0.0.1:$MASTER_PORT/apis/${API_VERSION}/status/test '"name":"test","stage":"Running"' 10
 
+    run_dm_worker $WORK_DIR/worker3 $WORKER3_PORT $cur/conf/dm-worker3.toml
+    check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER3_PORT
+
+    echo "isolate dm-worker4"
```
Comment (csuzhangxc) [inline tests/ha_cases/run.sh:198]: should we need to update the function name from `test_kill_worker` to another one with `isolate`?

File: dm/worker/config_test.go [hunk 9]
```diff
@@ -42,6 +42,7 @@ func (t *testConfigSuite) TestAdjustAddr(c *check.C) {
 	cfg.AdvertiseAddr = ""
 	c.Assert(terror.ErrWorkerHostPortNotValid.Equal(cfg.adjust()), check.IsTrue)
 
+	// TICASE-956
```
[the remark this answers is not in the document]
Comment (csuzhangxc) [reply dm/worker/config_test.go:45]: Can we public the test case design file?

[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply dm/worker/config_test.go:45]: I think it's ok

**Pull request merged**